core/src/main/java/org/mule/processor/AsyncInterceptingMessageProcessor.java

```diff
@@ -72,7 +72,7 @@ public WorkManager getWorkManager() throws MuleException
 
     public void start() throws MuleException
     {
-        if (workManager != null)
+        if (workManager != null && !workManager.isStarted())
```

Comment:
The problem was that before, the stop and dispose phases weren't been correctly applied inside the SPringRegistry.

If you look at com.mulesoft.mule.config.pool.MonitoredThreadPoolFactory#createPool, you'll see that the created workManager goes into the registry.

Because now lifecycle works correctly, if the flow is stopped, so is the workManager. So, if the flow is then restarted, the workManager must also be restarted because otherwise it will reject any incoming tasks. This is not a bug per se, but the solution to a bug that has always been there but never manifested, hidden behind the lifecycle chaos.